server/server.go

```diff
@@ -112,9 +117,26 @@ func (s *RpcEndPointServer) Start() {
 
 	<-notifier
 
+	s.SetUnhealthy()
+	s.logger.Info("http server stopping")
+
+	// Wait for 60 seconds for load balancer to pick up the unhealthy state or for SIGTERM to be sent again
+	select {
+	case <-time.After(60 * time.Second):
+	case <-notifier:
```

Comment:
**question:**

why would the `SIGTERM` sent for a second time?  I think if the OS/k8s gets impatient (that is, when its graceful shutdown timeout expires) it would then just send `SIGKILL` and that would be the end of story.

maybe convenient for the developer though..  (they can then hit Ctrl-C twice and avoid waiting for 60s when they want to terminate the server running locally).

Reply:
Yes, its for Ctrl+C. Not a big deal but for completeness.